zhaquirks/tuya/thermostat_88teujp.py

```diff
@@ -0,0 +1,251 @@
+                LOCAL_TEMP_REPORTED, decimal
+            )
+        elif tuya_cmd.command_id == SYSTEM_MODE_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                SYSTEM_MODE_REPORTED, decimal
+            )
+        elif tuya_cmd.command_id == BATTERY_STATE_COMMAND_ID:
+            self.endpoint.device.battery_bus.listener_event(BATTERY_REPORTED, decimal)
+        elif (
+            tuya_cmd.command_id >= 110 and tuya_cmd.command_id <= 129
+        ):  # Ignore historical data and schedules
+            pass
+        elif tuya_cmd.command_id == 362:  # Ignore away mode
+            pass
+        elif tuya_cmd.command_id == 364:  # Ignore auto/manual mode
+            pass
+        else:
+            _LOGGER.debug(
+                "Unknown command received: %d, %d",
+                tuya_cmd.command_id,
+                decimal,
+            )
+
+
+class PowerConfigurationCluster(LocalDataCluster, PowerConfiguration):
+    """Power configuration cluster."""
+
+    cluster_id = PowerConfiguration.cluster_id
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.battery_bus.add_listener(self)
+
+    def battery_reported(self, value):
+        """Handle reported battery state."""
+        _LOGGER.debug("reported battery alert: %d", value)
+        if value == 1:  # alert
+            self._update_attribute(
+                BATTERY_PERCENTAGE_REMAINING_ATTR, 0
+            )  # report 0% battery
+        else:
+            self._update_attribute(
+                BATTERY_PERCENTAGE_REMAINING_ATTR, 200
+            )  # report 100% battery
+
+
+class ThermostatCluster(LocalDataCluster, Thermostat):
+    """Thermostat cluster."""
+
+    cluster_id = Thermostat.cluster_id
+
+    _CONSTANT_ATTRIBUTES = {
+        MIN_HEAT_SETPOINT_ATTR: 500,
+        MAX_HEAT_SETPOINT_ATTR: 3000,
+        CTRL_SEQ_OF_OPER_ATTR: Thermostat.ControlSequenceOfOperation.Heating_Only,  # the device supports heating mode
+    }
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.thermostat_bus.add_listener(self)
+
+    def occupied_heating_setpoint_reported(self, value):
+        """Handle reported occupied heating setpoint state."""
+        self._update_attribute(OCCUPIED_HEATING_SETPOINT_ATTR, value * 10)
+        _LOGGER.debug("reported occupied heating setpoint: %d", value)
+
+    def local_temp_reported(self, value):
+        """Handle reported local temperature."""
+        self._update_attribute(LOCAL_TEMP_ATTR, value * 10)
+        _LOGGER.debug("reported local temperature: %d", value)
+
+    def system_mode_reported(self, value):
+        """Handle reported system mode."""
+        if value == 1:
+            self._update_attribute(SYSTEM_MODE_ATTR, Thermostat.SystemMode.Heat)
+            self._update_attribute(RUNNING_MODE_ATTR, Thermostat.RunningMode.Heat)
```

Comment:
In my PR I rely on the device to send the correct running_mode update when system_mode changes, but maybe your approach is better for flaky devices.

Reply:
I didn't find a way to get the running from the device

Reply:
Again, some devices send that running_mode very reliably, and some others seem not to, or not often enough. We are currently debugging some of those problems with MattsWeb